cmd/restic/cmd_cat.go

```diff
@@ -42,10 +42,12 @@ func runCat(gopts GlobalOptions, args []string) error {
 		return err
 	}
 
-	lock, err := lockRepo(gopts.ctx, repo)
-	defer unlockRepo(lock)
-	if err != nil {
-		return err
+	if !gopts.NoLock {
+		lock, err := lockRepo(gopts.ctx, repo)
+		defer unlockRepo(lock)
```

Comment:
The lint failure can be solved by doing

    defer func() { _ = unlockRepo(lock) }()

Though honestly, I think this check should be disabled. It leads to lots of "explicit error ignoring", which IMHO just hurts readability rather than improving the codebase.

Reply:
I looked around and the way I have it above is how it's replicated in all other instances of repo locking.

I agree that this check is not helpful.

Reply:
Alternatively, unlockRepo could print the error (with Warnf) instead of returning it.

Reply:
I also got the same error in #2876. I also vote for the `Warnf` solution!

Reply:
I think warnf would be a preferable solution.